### ISSUE TITLE
Nested Stage Failure in Parallel Execution Not Reflecting in Main Stage GUI Display

### ISSUE THREAD

**Issue by soumava11** [jenkinsci/pipeline-graph-view-plugin#515]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```Jenkins: 2.478
OS: Windows 10 - 10.0
Java: 21.0.3 - Amazon.com Inc. (OpenJDK 64-Bit Server VM)
---
Office-365-Connector:4.21.5
Parameterized-Remote-Trigger:3.2.0
active-directory:2.36
analysis-model-api:12.7.0
antisamy-markup-formatter:162.v0e6ec0fcfcf6
apache-httpcomponents-client-4-api:4.5.14-208.v438351942757
apache-httpcomponents-client-5-api:5.4-118.v199115451c4d
asm-api:9.7-33.v4d23ef79fcc8
audit-trail:361.v82cde86c784e
authentication-tokens:1.119.v50285141b_7e1
blueocean:1.27.16
blueocean-bitbucket-pipeline:1.27.16
blueocean-commons:1.27.16
blueocean-config:1.27.16
blueocean-core-js:1.27.16
blueocean-dashboard:1.27.16
blueocean-display-url:2.4.3
blueocean-events:1.27.16
blueocean-git-pipeline:1.27.16
blueocean-github-pipeline:1.27.16
blueocean-i18n:1.27.16
blueocean-jira:1.27.16
blueocean-jwt:1.27.16
blueocean-personalization:1.27.16
blueocean-pipeline-api-impl:1.27.16
blueocean-pipeline-editor:1.27.16
blueocean-pipeline-scm-api:1.27.16
blueocean-rest:1.27.16
blueocean-rest-impl:1.27.16
blueocean-web:1.27.16
bootstrap5-api:5.3.3-1
bouncycastle-api:2.30.1.78.1-248.ve27176eb_46cb_
branch-api:2.1178.v969d9eb_c728e
build-name-setter:2.4.3
build-time-blame:75.v44a_51cc97cdd-2.1.0
build-timeout:1.33
build-user-vars-plugin:178.v69708a_7a_dc17
caffeine-api:3.1.8-133.v17b_1ff2e0599
calendar-view:0.3.3
checks-api:2.2.1
cloudbees-bitbucket-branch-source:888.v8e6d479a_1730
cloudbees-folder:6.951.v5f91d88d76b_b_
command-launcher:115.vd8b_301cc15d0
commons-compress-api:1.26.1-2
commons-lang3-api:3.17.0-84.vb_b_938040b_078
commons-text-api:1.12.0-129.v99a_50df237f7
configuration-as-code:1850.va_a_8c31d3158b_
credentials:1378.v81ef4269d764
credentials-binding:681.vf91669a_32e45
customizable-header:135.vf8ce4237feb_c
dark-theme:479.v661b_1b_911c01
dashboard-view:2.517.v776a_b_811a_b_4e
data-tables-api:2.1.6-1
disable-job-button:1.v9db_352414f90
display-url-api:2.204.vf6fddd8a_8b_e9
durable-task:577.v2a_8a_4b_7c0247
echarts-api:5.5.1-1
eddsa-api:0.3.0-4.v84c6f0f4969e
email-ext:1844.v3ea_a_b_842374a_
emailext-template:1.5
embeddable-build-status:487.va_0ef04c898a_2
external-monitor-job:215.v2e88e894db_f8
favorite:2.221.v19ca_666b_62f5
flatpickr-api:4.6.13-5.v534d8025a_a_59
font-awesome-api:6.6.0-2
forensics-api:2.6.0
git:5.5.1
git-client:6.0.0
github:1.40.0
github-api:1.321-478.vc9ce627ce001
github-branch-source:1797.v86fdb_4d57d43
gson-api:2.11.0-41.v019fcf6125dc
handy-uri-templates-2-api:2.1.8-30.v7e777411b_148
htmlpublisher:1.36
http_request:1.19
instance-identity:185.v303dc7c645f9
integrity-plugin:2.6
ionicons-api:74.v93d5eb_813d5f
jackson2-api:2.17.0-379.v02de8ec9f64c
jakarta-activation-api:2.1.3-1
jakarta-mail-api:2.1.3-1
javadoc:280.v050b_5c849f69
javax-activation-api:1.2.0-7
javax-mail-api:1.6.2-10
jaxb:2.3.9-1
jdk-tool:80.v8a_dee33ed6f0
jenkins-design-language:1.27.16
jersey2-api:2.44-151.v6df377fff741
jira:3.13
jjwt-api:0.11.5-112.ve82dfb_224b_a_d
joda-time-api:2.13.0-85.vb_64d1c2921f1
jquery:1.12.4-1
jquery3-api:3.7.1-2
json-api:20240303-41.v94e11e6de726
json-path-api:2.9.0-58.v62e3e85b_a_655
junit:1300.v03d9d8a_cf1fb_
ldap:733.vd3700c27b_043
lockable-resources:1315.v4ea_8e5159ec8
log-parser:2.3.5
login-theme:146.v64a_da_cf70ea_6
mailer:472.vf7c289a_4b_420
mapdb-api:1.0.9-40.v58107308b_7a_7
matlab:2.14.0
matrix-auth:3.2.2
matrix-project:832.va_66e270d2946
metrics:4.2.21-451.vd51df8df52ec
mina-sshd-api-common:2.13.2-125.v200281b_61d59
mina-sshd-api-core:2.13.2-125.v200281b_61d59
nested-data-reporting:5.2.3
okhttp-api:4.11.0-181.v1de5b_83857df
opentelemetry:3.1391.vcb_a_a_b_9779d75
opentelemetry-api:1.40.0-36.v1e02b_b_4db_8f4
pam-auth:1.11
people-view:1.2
pipeline-build-step:540.vb_e8849e1a_b_d8
pipeline-github-lib:61.v629f2cc41d83
pipeline-graph-analysis:216.vfd8b_ece330ca_
pipeline-graph-view:349.veda_b_e2366d99
pipeline-groovy-lib:730.ve57b_34648c63
pipeline-input-step:495.ve9c153f6067b_
pipeline-milestone-step:119.vdfdc43fc3b_9a_
pipeline-model-api:2.2214.vb_b_34b_2ea_9b_83
pipeline-model-definition:2.2214.vb_b_34b_2ea_9b_83
pipeline-model-extensions:2.2214.vb_b_34b_2ea_9b_83
pipeline-rest-api:2.34
pipeline-stage-step:312.v8cd10304c27a_
pipeline-stage-tags-metadata:2.2214.vb_b_34b_2ea_9b_83
pipeline-stage-view:2.34
pipeline-utility-steps:2.17.0
plain-credentials:183.va_de8f1dd5a_2b_
plot:2.1.12
plugin-util-api:5.1.0
prism-api:1.29.0-17
pubsub-light:1.18
rebuild:332.va_1ee476d8f6d
resource-disposer:0.23
role-strategy:743.v142ea_b_d5f1d3
scm-api:696.v778d637b_a_762
script-security:1362.v67dc1f0e1b_b_3
simple-theme-plugin:196.v96d9592f4efa_
snakeyaml-api:2.3-123.v13484c65210a_
sse-gateway:1.27
ssh-credentials:343.v884f71d78167
sshd:3.330.vc866a_8389b_58
structs:338.v848422169819
subversion:1275.va_7b_014f3fc2c
theme-manager:262.vc57ee4a_eda_5d
timestamper:1.27
token-macro:400.v35420b_922dcb_
trilead-api:2.147.vb_73cc728a_32e
variant:60.v7290fc0eb_b_cd
warnings-ng:11.9.0
workflow-aggregator:600.vb_57cdd26fdd7
workflow-api:1336.vee415d95c521
workflow-basic-steps:1058.vcb_fc1e3a_21a_9
workflow-cps:3964.v0767b_4b_a_0b_fa_
workflow-durable-task-step:1371.vb_7cec8f3b_95e
workflow-job:1436.vfa_244484591f
workflow-multibranch:795.ve0cb_1f45ca_9a_
workflow-scm-step:427.v4ca_6512e7df1
workflow-step-api:678.v3ee58b_469476
workflow-support:926.v9f4f9b_b_98c19
ws-cleanup:0.46
  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Windows

### Reproduction steps

To reproduce the issue- create parallel stages. Screenshot attached. 

In the following image, one of the nested stages failed inside a parallel stage. 
Image 1: ![image](https://github.com/user-attachments/assets/65e262b5-011c-4ace-a9f1-a23c61197ec0)

However, in the top level of the job dashboard- the nested stage shows 'SUCCESS'. 
Image 2: ![image](https://github.com/user-attachments/assets/0b9e8a19-e2a0-4a97-8ee2-c1b56db1d6dd)

Clicking on Image 2 takes me to Image 1, which shows the parallel nested stage 'FAILED'. 

### Expected Results

 ![image](https://github.com/user-attachments/assets/0b9e8a19-e2a0-4a97-8ee2-c1b56db1d6dd) 

The parallel main stage should also be marked as 'FAILED' since one stage inside failed and the overall pipeline failed. 

### Actual Results

 ![image](https://github.com/user-attachments/assets/0b9e8a19-e2a0-4a97-8ee2-c1b56db1d6dd) 

The parallel main stage shows 'SUCCESS'.

**Comment by timja** [comment]
Duplicate of https://github.com/jenkinsci/pipeline-graph-view-plugin/issues/327
